From: Richard Goedeken <Richard@fascinationsoftware.com>
Date: Thu, 1 Oct 2015 20:32:35 -0700
Subject: added specific config for Logitech F310 from Marcus Dean Adams on
 google group,
 who suggested that this config is better than the old mapping which used the
 same config as other XBox Clones

Origin: upstream, https://github.com/mupen64plus/mupen64plus-input-sdl/commit/bcf6892ca613ae6471ddb01c15792878f30bad67
---
 data/InputAutoCfg.ini | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/data/InputAutoCfg.ini b/data/InputAutoCfg.ini
index 5efd57f..3111c97 100644
--- a/data/InputAutoCfg.ini
+++ b/data/InputAutoCfg.ini
@@ -530,6 +530,31 @@ Rumblepak switch = button(7)
 X Axis = axis(0-,0+)
 Y Axis = axis(1-,1+)
 
+[Logitech Gamepad F310]
+plugged = True
+plugin = 2
+mouse = False
+AnalogDeadzone = 4096,4096
+AnalogPeak = 32768,32768
+DPad R = hat(0 Right)
+DPad L = hat(0 Left)
+DPad D = hat(0 Down)
+DPad U = hat(0 Up)
+Start = button(7)
+Z Trig = button(5)
+B Button = button(2)
+A Button = button(0)
+C Button R = axis(3+)
+C Button L = axis(3-)
+C Button D = axis(4+)
+C Button U = axis(4-)
+R Trig = axis(5-)
+L Trig = axis(2-)
+Mempak switch = button(1)
+Rumblepak switch = button(3)
+X Axis = axis(0-,0+)
+Y Axis = axis(1-,1+)
+
 [Logitech Gamepad F710]
 plugged = True
 plugin = 2
@@ -1407,7 +1432,6 @@ Y Axis = axis(1-,1+)
 [Thrustmaster, Inc. Controller]
 [Logitech Xbox Cordless Controller]
 [Logitech Compact Controller for Xbox]
-[Logitech Gamepad F310]
 [Mad Catz Controller (unverified)]
 [InterAct 'PowerPad Pro' X-Box pad (Germany)]
 [Mad Catz Control Pad]
